### DISCUSSION TITLE
Very slow query when using "IN" with multiple values

### DISCUSSION THREAD

**Discussion by n-e** [duckdb/duckdb#11009] [Q&A] [answered]
I have a table such as this one:
```
┌───────┬──────────────────────────────────────────────────────────────────────────────────────────┐
│  id   │                                           data                                           │
│ int64 │                                         varchar                                          │
├───────┼──────────────────────────────────────────────────────────────────────────────────────────┤
│     1 │ 79ac3e385465e04bf4a4b2a8b93dc8542bc7e455eb21219ae9d0f1e66d06567fb58bc2bf716590a30a0658…  │
│     2 │ 93ab19f622558e46148cddbb11dc4cc890a795215929ca31d7d222b1b6f4be4926aed9123089f003ec3b36…  │
└───────┴──────────────────────────────────────────────────────────────────────────────────────────┘
```

It can be generated with this command:
```sh
cat /dev/urandom|xxd -p -c 256|head -1000000|jq -R '{id: input_line_number, data:.}'|duckdb -c "create or replace table t as select * from read_json('/dev/stdin')" demo.duck
```

The following query takes 1.7ms : 
```
select id,data from t where id in (1000);
```

However this one takes 310ms:
```
select id,data from t where id in (1000,800000);
```

I would expect it to be much faster, since running the first query twice with the two values would take about 3.4ms.

Any ideas on how to make this faster? In my real use case, the in clause would contain about 10-20 values.

**Comment by soerenwolfers** [comment, marked as answer]
The query plans are different (as you can see by prepending `EXPLAIN` to your query). If you used multiple consecutive values, duckdb would keep using the former plan.

Since duckdb is normally not designed for low latency, I don't think the developers will spend much time micro-optimizing the non-consecutive plan to run multiple consecutive plans for low numbers of entries. 

The good news is that once you're on the second plan, the runtime stays essentially constant until you get to 10k entries.

**Reply by n-e** [reply]
Thanks for the input! I was doing this query in duckdb to keep things simple (so I have all the data in the same place) although it isn’t super suited for it. I’ll move this table to another database.

**Comment by soerenwolfers** [comment]
Somewhat related: https://github.com/duckdb/duckdb/issues/4295
